### ISSUE TITLE
fail_on_error flag is not working as expected

### ISSUE THREAD

**Issue by dmchandru** [reviewdog/action-eslint#73]
My workflow config file as follows
```
jobs:
  eslint:
    name: runner / eslint
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - name: eslint
        uses: reviewdog/action-eslint@v1
        with:
          reporter: github-pr-review # Change reporter.
          workdir: 'src/'
          fail_on_error: true
```
Initially the runner was reported the status check as failed if the PR code have any lint issues. But now the status check seems to passed even if the code have lint issues.

**Comment by plata** [comment]
Can confirm:
- [yml](https://github.com/PhoenicisOrg/scripts/blob/ced4e14dc14d98ff5e4e4f3a5ce1d1cfeb59f0f5/.github/workflows/pr-review.yml#L22)
- commit: https://github.com/PhoenicisOrg/scripts/pull/1231/commits/0e7c1ef35734b446a9a435e1701d52a84fc63f17

You can see that the error got reported but the action still passed. I only noticed it because I'm still using Codacy as well.
